packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -356,43 +408,75 @@ abstract class RouteBaseConfig {
     final InterfaceElement classElement = typeParamType.element;
 
     final RouteBaseConfig value;
-    if (isShellRoute) {
-      value = ShellRouteConfig._(
-        routeDataClass: classElement,
-        parent: parent,
-        navigatorKey: _generateNavigatorKeyGetterCode(
-          classElement,
-          keyName: r'$navigatorKey',
-        ),
-        parentNavigatorKey: _generateNavigatorKeyGetterCode(
-          classElement,
-          keyName: r'$parentNavigatorKey',
-        ),
-      );
-    } else {
-      final ConstantReader pathValue = reader.read('path');
-      if (pathValue.isNull) {
-        throw InvalidGenerationSourceError(
-          'Missing `path` value on annotation.',
-          element: element,
+    switch (typeName) {
+      case 'TypedShellRoute':
+        value = ShellRouteConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
         );
-      }
+        break;
+      case 'TypedStatefulShellRoute':
+        value = StatefulShellRouteConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
+        );
+        break;
+      case 'TypedStatefulShellBranch':
+        value = StatefulShellBranchConfig._(
+          routeDataClass: classElement,
+          parent: parent,
+          navigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$navigatorKey',
+          ),
+          parentNavigatorKey: _generateNavigatorKeyGetterCode(
+            classElement,
+            keyName: r'$parentNavigatorKey',
+          ),
+        );
+        break;
+      default:
```

Comment:
why not use `case 'TypedGoRoute':`?
it should throw if it is anything else.

Reply:
done